File: cases/taint/java/element-object-positive/ElementObjectPositive.java
package dataflowbench.taint;

final class ElementObjectPositive {
    static final class Item {
        int value;
    }

    static int dfb_source() { // DFB-SOURCE: element-object-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: element-object-sink

    static void run() {
        Item[] items = new Item[] {new Item(), new Item()};
        items[0].value = dfb_source(); // DFB-WITNESS: element-object-store
        items[1].value = 0;
        dfb_sink(items[0].value);
    }
}

File: cases/taint/kotlin/element-object-positive/ElementObjectPositive.kt
package dataflowbench

object ElementObjectPositive {
    class Item {
        var value: String = "clean"
    }

    fun dfb_source(): String { // DFB-SOURCE: element-object-input
        return "tainted"
    }

    fun dfb_sink(value: String) {} // DFB-SINK: element-object-sink

    fun run() {
        val items = arrayOf(Item(), Item())
        items[0].value = dfb_source() // DFB-WITNESS: element-object-store
        items[1].value = "clean"
        dfb_sink(items[0].value)
    }
}
